### PR TITLE
Documenting the IgxAvatarComponent

### LINKED ISSUE #1441: Document the IgxAvatarComponent
The IgxAvatarComponent needs to be documented.

### PR DESCRIPTION
**Pull request by HristoP96**
Closes #1441

### CODE REVIEW

File: projects/igniteui-angular/src/lib/avatar/avatar.component.ts [hunk 1]
```diff
@@ -37,58 +37,157 @@ export enum Size {
...
+ * If the avatar has an icon, it will have the 'iconTemplate'.
+ *
+ * @memberof IgxAvatarComponent
+ */
+@ViewChild('iconTemplate', { read: TemplateRef })
     protected iconTemplate: TemplateRef<any>;
-
-    @HostBinding('attr.aria-label')
+/**
+ * Indicates that the avatar's attribute aria-label is 'avatar'.
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/avatar/avatar.component.ts]: Add a code snippet for the inputs/methods/outputs

File: projects/igniteui-angular/src/lib/avatar/avatar.component.ts [hunk 2]
```diff
@@ -37,58 +37,202 @@ export enum Size {
...
+     * <igx-avatar role="img"></igx-avatar>
+     * ```
+     *
+     * @memberof IgxAvatarComponent
+     */
     @HostBinding('attr.role')
     public role = 'img';
-
+    /**
+     * Initializes the avatar's class.
```
Comment (hanastasov) [inline projects/igniteui-angular/src/lib/avatar/avatar.component.ts]: This is only a getter - it returns the avatar's class, does not set it or initialize it. Therefore the code asnippet should be how to get it in typescript.

File: projects/igniteui-angular/src/lib/avatar/avatar.component.ts [hunk 3]
```diff
@@ -111,18 +263,32 @@ export class IgxAvatarComponent implements OnInit, AfterViewInit {
...
+     * @memberof IgxAvatarComponent
+     */
     public ngOnInit() {
         this.roleDescription = this.getRole();
     }
 
+    /**
+     *@hidden
+     *
+     * @memberof IgxAvatarComponent
```
Comment (hanastasov) [inline projects/igniteui-angular/src/lib/avatar/avatar.component.ts]: * @memberof IgxAvatarComponent - remove it, if the property/method/input is hidden.

File: projects/igniteui-angular/src/lib/avatar/avatar.component.ts [hunk 4]
```diff
@@ -37,58 +37,202 @@ export enum Size {
...
+     * ```html
+     * <igx-avatar aria-label="avatar"></igx-avatar>
+     * ```
+     * @memberof IgxAvatarComponent
+     */
     @HostBinding('attr.aria-label')
     public ariaLabel = 'avatar';
-
+    /**
+     * Indicates that the avatar's attribute role is 'img'.
```
Comment (hanastasov) [inline projects/igniteui-angular/src/lib/avatar/avatar.component.ts]: Returns, not indicates. The code snippet should be typescript

File: projects/igniteui-angular/src/lib/avatar/avatar.component.ts [hunk 5]
```diff
@@ -133,6 +299,15 @@ export class IgxAvatarComponent implements OnInit, AfterViewInit {
         }
     }
 
+    /**
+     * Returns the url of the image.
+     *
+     * ```typescript
+     * duplicateAvatar.src = this.avatar.getSrcUrl();
```
Comment (hanastasov) [inline projects/igniteui-angular/src/lib/avatar/avatar.component.ts]: duplicateAVatar does not make sense here.

**Pull request merged**